language-server/src/main/kotlin/tools/samt/ls/SamtTextDocumentService.kt

```diff
@@ -91,10 +93,11 @@ class SamtTextDocumentService(private val workspace: SamtWorkspace) : TextDocume
             val relevantFileNode = relevantFileInfo.fileNode ?: return@supplyAsync emptyList()
             val token = relevantFileInfo.tokens.findAt(params.position) ?: return@supplyAsync emptyList()
 
-            val (_, files, globalPackage) = workspace.getFolderSnapshot(path) ?: return@supplyAsync emptyList()
-            if (globalPackage == null) return@supplyAsync emptyList()
+            val (_, files, semanticModel) = workspace.getFolderSnapshot(path) ?: return@supplyAsync emptyList()
+            if (semanticModel == null) return@supplyAsync emptyList()
 
-            val typeLookup = SamtDeclarationLookup.analyze(relevantFileNode, globalPackage.resolveSubPackage(relevantFileInfo.fileNode.packageDeclaration.name))
+            val globalPackage = semanticModel.global
+            val typeLookup = SamtDeclarationLookup.analyze(relevantFileNode, globalPackage.resolveSubPackage(relevantFileInfo.fileNode.packageDeclaration.name), userMetadata = semanticModel.userMetadata)
```

Comment:
Why add a name to only the last argument? Maybe add names to all arguments and spread call onto multiple lines for better readability